### ISSUE TITLE
Operator keeps recreating sts / restarting pods (environment does not match)

### ISSUE THREAD

**Issue by bumarcell** [stephan2012/postgres-log-exporter#1]
Thanks a lot for your work, it has benefited us a lot.

We do have one problem tho, which might be caused by some operator bug, but I thought I would ask here;
After having added `log-exporter` sidecare (including the `NODE_NAME` env from fieldRef, the operator keeps recreating the statefulset with every reconciliation after show the following log:
`level=info msg="reason: new statefulset containers's log-exporter (index 2) environment does not match the current one" cluster-name=test-postgres pkg=cluster`

Removing the env definition from sidecar stops this behavior. 

Have you seen this before?

**Comment by stephan2012** [comment]
No, I haven't seen this behavior before. But I'm not entirely up to date with the recent operator release. I'm keeping this issue in mind when updating.

**Comment by bumarcell** [comment]
Not the latest, we use Zalando operator version 1.11.0 patched with customizable securityContext for postgres containers.
